### DISCUSSION TITLE
Client queries 127.0.1.1 - how to block?

### DISCUSSION THREAD

**Discussion by ghost** [AdguardTeam/AdGuardHome#4753] [Q&A]
My Apple TV queries my Raspberry Pi server hostname (BOB) and AGH responds with REWRITE 127.0.1.1. I think that is the correct response because "/etc/hosts" does specify BOB for 127.0.1.1. Is it possible to block/blackhole response to server hostname? I keep adding BOB to custom rules, but that doesn't work, probably because BOB is specified in "/etc/hosts" file. Linux requires hostname to be specified under 127.0.1.1, but I want the response blackholed.  How would even Apple TV know to query the hostname? Does it leak somewhere?

**Comment by ainar-g** [comment]
Hello.

 >  Is it possible to block/blackhole response to server hostname? I keep adding BOB to custom rules, but that doesn't work, probably because BOB is specified in "/etc/hosts" file.  Linux requires hostname to be specified under 127.0.1.1, but I want the response blackholed.

It is not currently possible to exclude a hostname from the `/etc/hosts` file.  But you can disable the `/etc/hosts` processing completely, if you don't really use it, by setting `clients.runtime_sources.hosts` to `false`.

Another way is to add both `BOB` and `1.1.0.127.in-addr.arpa` to the *Disallowed domains* field on the *Settings → DNS settings* page.

 >  How would even Apple TV know to query the hostname?  Does it leak somewhere?

Apple TV could simply go through its network neighbourhood (e.g. `arp -a`), making rDNS queries for all IP addresses it finds.  Probably to show human-readable names somewhere in the UI.

**Comment by ghost** [comment]
Thanks!

What are performance differences between blackholing undesired domains VS disallowing undesired domains? I assume disallowing domains returns "No Response" / "Time Out" response. "Time Out" responses usually take longer than 0.0.0.0 responses, but 0.0.0.0 responses indicate that there is a blackhole, while "Time Out" responses indicate that DNS server not available.

**Reply by ainar-g** [reply]
Depends on what you mean by “performance”.  The client will obviously hang for a longer time, so that is generally not a good choice for e.g. browsers, but AdGuard Home won't waste the resources that it would have otherwise wasted if it were to actually process the query.

**Comment by ghost** [comment]
Does AGH query show when resolution is blocked due to client exceeding set rate limit?
